src/bucket/test/BucketIndexTests.cpp

```diff
@@ -413,4 +431,85 @@ TEST_CASE("loadPoolShareTrustLinesByAccountAndAsset does not load shadows",
 
     testAllIndexTypes(f);
 }
+
+TEST_CASE("serialize bucket indexes", "[bucketindex]")
+{
+    Config cfg(getTestConfig());
+
+    // First 3 levels individual, last 3 range index
+    cfg.EXPERIMENTAL_BUCKETLIST_DB_INDEX_CUTOFF = 1;
+    cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+
+    auto test = BucketIndexTest(cfg);
+    test.buildGeneralTest();
+
+    auto& bm = test.getBM();
+    for (auto const& bucketHash : bm.getBucketListReferencedBuckets())
+    {
+        if (isZero(bucketHash))
+        {
+            continue;
+        }
+
+        // Check if index files are saved
+        auto indexFilename = bm.bucketFilename(bucketHash, /*isIndex=*/true);
+        REQUIRE(fs::exists(indexFilename));
+
+        auto bucket = bm.getBucketByHash(bucketHash);
+        auto onDiskIndex =
+            BucketIndex::load(bm, indexFilename, bucket->getSize());
+        REQUIRE(onDiskIndex);
+
+        auto& originalIndex = bucket->getIndexForTesting();
+        REQUIRE((originalIndex == *onDiskIndex));
+    }
+}
+
+TEST_CASE("on disk index out of date", "[bucketindex]")
+{
+    // Generate a one level BucketList with an individual index
+    Config cfg(getTestConfig(0, Config::TESTDB_ON_DISK_SQLITE));
+    cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+    auto test = BucketIndexTest(cfg, 1);
+    test.buildGeneralTest();
+
+    auto buckets = test.getBM().getBucketListReferencedBuckets();
+    for (auto const& bucketHash : buckets)
+    {
+        if (isZero(bucketHash))
+        {
+            continue;
+        }
+
+        // Check if index files are saved
+        auto indexFilename =
+            test.getBM().bucketFilename(bucketHash, /*isIndex=*/true);
+        REQUIRE(fs::exists(indexFilename));
+    }
+
+    // Restart app with different config to test that indexes created with
+    // different config settings are not loaded from disk
+    cfg.EXPERIMENTAL_BUCKETLIST_DB_INDEX_CUTOFF = 0;
+    cfg.EXPERIMENTAL_BUCKETLIST_DB_INDEX_PAGE_SIZE_EXPONENT = 10;
+
+    // When the app restarts, indexes are loaded from disk. If an index file is
+    // outdated, it is deleted. By setting persist index to false, no new
+    // indexes will be written, so we can check the existence of the index file
+    // to see if we properly dropped the outdated files
+    cfg.EXPERIMENTAL_BUCKETLIST_DB_PERSIST_INDEX = false;
+
+    test.restartWithConfig(cfg);
+    for (auto const& bucketHash : buckets)
+    {
+        if (isZero(bucketHash))
+        {
+            continue;
+        }
+
+        // Check if index file has been deleted
+        auto indexFilename =
```

Comment:
Should we verify here that index files with new configs were created and are valid?

Reply:
(Marking resolved as Marta is on vacation and I think this has been addressed)